README.md

```diff
@@ -6,3 +6,218 @@
+    await nameSchema.validate(v);
+    return v;
+  },
+});
+
+const lastNameAtom = atomWithValidate('', {
+  validate: async (v) => {
+    await nameSchema.validate(v);
+    return v;
+  },
+});
+
+const formGroup = validateAtoms(
+  {
+    firstName: firstNameAtom,
+    lastName: lastNameAtom,
+  },
+  (values) => {
+    if (values.firstName.length + values.lastName.length > 15) {
+      throw new Error("Overall name can't be longer than 15 characters");
+    }
+  },
+);
+
+const Form = () => {
+  const [firstName, setFirstName] = useAtom(firstNameAtom);
+  const [lastName, setLastName] = useAtom(lastNameAtom);
+  const [formState] = useAtom(formGroup);
+
+  return (
+    <div>
+      <label>
+        firstName
+        <input
+          value={firstName.value}
+          onChange={(e) => setFirstName(e.target.value)}
+        />
+        <span>{firstName.isValid && 'Valid'}</span>
+        <span>{!firstName.isValid && `${email.error}`}</span>
+      </label>
+
+      <label>
+        lastName
+        <input
+          value={lastName.value}
+          onChange={(e) => setLastName(e.target.value)}
+        />
+        <span>{lastName.isValid && 'Valid'}</span>
+        <span>{!lastName.isValid && `${email.error}`}</span>
+      </label>
+
+      <button
+        disabled={!formState.isValid || !firstName.isValid || !lastName.isValid}
+      >
+        Submit
+      </button>
+      {
+        // or
+        // <button disabled={!(formState.isValid && firstName.isValid && lastName.isValid)}>Submit</button>
+      }
+      <span>{!formState.isValid && `${formState.error}`}</span>
+    </div>
+  );
+};
+```
+
+## API
+
+#### `atomWithValidate`
+
+```js
+atomWithValidate(initialValue, options);
+```
+
+| param              | description                                                 | default     |
+| ------------------ | ----------------------------------------------------------- | ----------- |
+| `initialValue`     | The initial value of the atom                               | `undefined` |
+| `options.validate` | function returning value or `throw`s an error               | `undefined` |
+|                    | receives the current value of the atom                      |             |
+| `options.areEqual` | function to compare the initial value to the changed values | `Object.is` |
+
+#### `validateAtoms`
+
+```js
+atomWithValidate(atomGroup, validator);
+```
+
+| param       | description                                        | default     |
+| ----------- | -------------------------------------------------- | ----------- |
+| `atomGroup` | an object of `atomsWithValidate` atoms             | `undefined` |
+|             | the keys of the atomGroup are used as names/labels |             |
+|             | when passed to the `validator` function            |             |
+| `validator` | function returning void or `throw`s an error       | `undefined` |
+|             | receives the atomGroup's values with the same keys |             |
```

Comment:
Would you like to describe about async validator? Same for options.validate in atomWithValidate.

Reply:
Sure, I'll add in a bit more detail about the validators

Reply:
Updated

Reply:
feel free to merge it.